llvm/lib/IR/RepoTicket.cpp

```diff
@@ -116,7 +116,39 @@ static GODigestState accumulateGOFinalDigestOrInitialDigestAndGetGODigestState(
   return accumulateGOInitialDigestAndGetGODigestState(GO, GOHash, GOIMap);
 }
 
-// Update the GO's hash value by adding the hash of its dependents.
+static bool updateDigestUseContributions(const GlobalObject *GO, MD5 &GOHash,
+                                         const GOVec &Contributions,
+                                         GOInfoMap &GOIMap) {
+  for (const GlobalObject *const G : Contributions) {
+    assert(isa<GlobalVariable>(GO) &&
+           "Only global variables can have contributions!");
+    assert(isa<llvm::Function>(G) && "All contributions are functions!");
+    GOHash.update(GOIMap[G].InitialDigest.Bytes);
+  }
+  return !Contributions.empty();
+}
+
+template <typename Function>
```

Comment:
Would be nice to have a comment saying what this function does.

Reply:
Added, thanks.